custom_components/xiaomi_miio_airpurifier/fan.py

```diff
@@ -745,6 +757,14 @@
 SERVICE_SET_DRY_ON = "fan_set_dry_on"
 SERVICE_SET_DRY_OFF = "fan_set_dry_off"
 
+# Airfresh T2017
+SERVICE_SET_FAVORITE_SPEED = "fan_set_favorite_speed"
+SERVICE_SET_DISPLAY_ON = "fan_set_display_on"
+SERVICE_SET_DISPLAY_OFF = "fan_set_display_off"
+SERVICE_SET_PTC_LEVEL = "fan_set_ptc_level"
+SERVICE_SET_POWER_ON = "fan_set_power_on"
```

Comment:
`fan_set_power_on` and `fan_set_power_on` isn't needed because the `fan` entity has an on/off state. Just use the `fan.turn_on` and `fan.turn_off` service to power the device on/off.